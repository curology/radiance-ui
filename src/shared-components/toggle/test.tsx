import React from 'react';
import { render, userEvent } from 'src/tests/testingLibraryHelpers';

import { Toggle } from './index';

describe('<Toggle />', () => {
  const labelText = 'Label Text';
  describe('UI snapshot', () => {
    it('renders the component', () => {
      const { container } = render(
        <Toggle checked={false} label={labelText} onChange={() => undefined} />,
      );

      expect(container.firstElementChild).toMatchSnapshot();
    });
  });

  describe('when label is undefined', () => {
    it('does not render a label component', () => {
      const { container } = render(
        <Toggle checked={false} onChange={() => undefined} />,
      );
      // TODO: Make <Label /> an actual label element and use @testing-library/react selector
      expect(container.querySelectorAll('span')).toHaveLength(0);
    });
  });

  describe('when label is a string', () => {
    it('renders a text component', () => {
      const { getByText } = render(
        <Toggle checked={false} label={labelText} onChange={() => undefined} />,
      );

      expect(getByText(labelText)).toBeTruthy();
    });
  });

  describe('when checkbox is clicked', () => {
    it('fires onChange function with correct argument when function exists', () => {
      const spy = jest.fn();
      const { getByRole } = render(<Toggle checked={false} onChange={spy} />);

      userEvent.click(getByRole('checkbox'));
      expect(spy).toHaveBeenCalledWith(false);
    });
  });
});
