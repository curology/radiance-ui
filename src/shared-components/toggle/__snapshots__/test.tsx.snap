// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Toggle /> UI snapshot renders the component 1`] = `
.emotion-0 {
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-flow: row nowrap;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  padding: 1rem 0;
}

.emotion-0>input {
  display: none;
}

.emotion-2 {
  color: #524D6E;
  margin: 0;
  font-size: 1rem;
  line-height: 1.5rem;
  text-align: left;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  margin-right: 0.5rem;
}

.emotion-4>div:first-of-type>input:focus {
  box-shadow: 0px 0px 0px 2px #FFFFFF,0px 0px 0px 4px #332E54;
  outline: none;
  border-radius: 12px;
}

<div
  class="emotion-0 emotion-1"
>
  <span
    class="emotion-2 emotion-3"
  >
    Label Text
  </span>
  <div
    class="emotion-4 emotion-5"
  >
    <div
      style="display: flex; width: 52px; align-items: center; justify-content: flex-start; position: relative; cursor: pointer; background-color: transparent; border: 0px; padding: 0px; user-select: none;"
    >
      <div
        style="width: 40px; height: 24px; padding: 0px; border-radius: 100px; display: flex; align-items: center; justify-content: center; background-color: rgb(225, 224, 230);"
      >
        <div
          style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgb(250, 250, 250); margin-top: auto; margin-bottom: auto; line-height: 0; opacity: 0; width: 26px; height: 20px; left: 4px;"
        />
        <div
          style="font-size: 11px; display: flex; align-items: center; justify-content: center; font-family: 'Helvetica Neue', Helvetica, sans-serif; position: relative; color: rgba(255, 255, 255, 0.6); bottom: 0px; margin-top: auto; margin-bottom: auto; padding-right: 5px; line-height: 0; width: 26px; height: 20px; opacity: 1;"
        />
      </div>
      <div
        style="position: absolute; height: 100%; top: 0px; left: 0px; display: flex; flex: 1; align-self: stretch; align-items: center; justify-content: flex-start;"
      >
        <div
          style="width: 22px; height: 22px; display: flex; align-self: center; box-shadow: none; border-radius: 50%; box-sizing: border-box; border: 1px solid #ededf0; background: rgb(255, 255, 255); background-color: rgb(255, 255, 255); position: relative; left: 1px;"
        />
      </div>
      <input
        style="margin: 0px; height: 24px; width: 40px; position: absolute;"
        type="checkbox"
        value="false"
      />
    </div>
  </div>
</div>
`;
